### DISCUSSION TITLE
Incorrect use of HttpClient

### DISCUSSION THREAD

**Discussion by Juka117** [olsh/curl-to-csharp#26] [General]
Putting the HttpClient in a using in this case is not the best way to go. If you put your curl to C# output into a timer with 50ms delay it will break after 15 seconds or so.

Instead you should apply the practice in this article:
https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

Otherwise, super tool mind you. Thank you!

**Comment by olsh** [comment]
Hi @Juka117,

Thanks for the feedback. 👍 

I'm aware of the problem. 
The tool was created to produce a minimum piece of code which can be compiled or run with LinqPad or something. I can remove the using statement and leave a comment like `inject your singleton HttpClient here` but in some cases, we need to create HttpClient with custom HttpHandler (for example with -x Proxy parameter).

If you have any suggestions on how I can improve the situation, please let me know.

**Comment by vladd** [comment]
Perhaps something like this could be generated:

```
// in production code, don't destroy the HttpClient through using, but better reuse an existing instance
using var httpClient = new HttpClient();
using (var request = new HttpRequestMessage(new HttpMethod("POST"), "http://example/"))
{
    request.Content = new StringContent("test");
    request.Content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/x-www-form-urlencoded"); 
    var response = await httpClient.SendAsync(request);
}
```

I used a simplified form of `using` in order to avoid extra indentation (which should anyway be removed in the typical usage).

**Reply by olsh** [reply]
Done, feel free to improve the comment
https://github.com/olsh/curl-to-csharp/blob/24f42416b6a8664d62698346a296155f682866e4/src/CurlToCSharp/Services/ConverterService.cs#L516-L518
